pageserver/src/walreceiver/connection_manager.rs

```diff
@@ -735,8 +756,11 @@ impl WalreceiverState {
             }
         });
 
-        for node_id in node_ids_to_remove {
-            self.wal_connection_retries.remove(&node_id);
+        if !node_ids_to_remove.is_empty() {
+            info!("Safekeeper nodes {node_ids_to_remove:?} did not send events for over {lagging_wal_timeout:?}, stopping their retries.");
+            for node_id in node_ids_to_remove {
+                self.wal_connection_retries.remove(&node_id);
+            }
```

Comment:
Would it be better to put the `info!()` log line inside the loop, one line per node id that is removed?
Makes the log more uniform.